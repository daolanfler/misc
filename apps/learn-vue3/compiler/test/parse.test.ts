import { expect, it, test } from 'vitest';
import { parse, tokenize } from '../parse';

test('tokenize works correctly', () => {
  const str = '<div><p>Vue</p><p>Template</p></div>';
  expect(tokenize(str)).toMatchSnapshot();
});

it('should parse to a template ast tree', () => {
  const template = '<div><p>Vue</p><p>Template</p></div>'
  expect(parse(template)).toMatchSnapshot();
});
