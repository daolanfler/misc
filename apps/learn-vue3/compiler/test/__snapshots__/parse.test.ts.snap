// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should parse to a template ast tree 1`] = `
{
  "children": [
    {
      "children": [
        {
          "children": [
            {
              "content": "Vue",
              "type": "Text",
            },
          ],
          "tag": "p",
          "type": "Element",
        },
        {
          "children": [
            {
              "content": "Template",
              "type": "Text",
            },
          ],
          "tag": "p",
          "type": "Element",
        },
      ],
      "tag": "div",
      "type": "Element",
    },
  ],
  "type": "Root",
}
`;

exports[`tokenize works correctly 1`] = `
[
  {
    "name": "div",
    "type": "tag",
  },
  {
    "name": "p",
    "type": "tag",
  },
  {
    "content": "Vue",
    "type": "text",
  },
  {
    "name": "char",
    "type": "tagEnd",
  },
  {
    "name": "p",
    "type": "tag",
  },
  {
    "content": "Template",
    "type": "text",
  },
  {
    "name": "char",
    "type": "tagEnd",
  },
  {
    "name": "char,i,v",
    "type": "tagEnd",
  },
]
`;
